roles/edpm_tripleo_cleanup/tasks/main.yml

```diff
@@ -0,0 +1,42 @@
+---
+# Copyright 2024 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Set list of tripleo services
+  tags:
+    - adoption
+  ansible.builtin.set_fact:
+    tripleo_services: "{{ edpm_old_tripleo_services }}"
+
+- name: Discover tripleo services
+  tags:
+    - adoption
+  when: tripleo_services | length == 0
+  block:
+    - name: Get all services
+      ansible.builtin.service_facts:
+    - name: Filter for tripleo services
+      ansible.builtin.set_fact:
+        tripleo_services: "{{ ansible_facts.services.keys() | select('contains', 'tripleo') }}"
+
+- name: Stop and disable tripleo services
+  tags:
+    - adoption
+  become: true
+  ansible.builtin.systemd_service:
+    name: "{{ item }}"
+    state: stopped
+    enabled: false
+  loop: "{{ tripleo_services }}"
```

Comment:
I think we also want to delete the service definition so that nobody can start those services up after the adoption.

Reply:
I'd rather not delete them completely. After all, let's give our customers some credit and assume they won't be enabling services for now reason.

But how about just masking them. That would make clear that these aren't supposed to run without limiting people too much.

Reply:
For what case we want to keep those tripleo systemd units to exists after adoption is done and there is no revert possible? If those are not used would be good to cleanup those completely otherwise those references will live forever on the systems.

Reply:
It's literally just a file that doesn't do anything and can't even be enabled without further manual intervention. 
Once the package that installed them is removed, which is something we should also automate at some point as part of adoptions, the file will go as well. Assuming it was installed with a package, which I believe is our standard approach.

Reply:
which package installs those? i don't think any package involved in those tripleo* systemd files, if it does those packages should be fixed.

Reply:
I would assume they belong to one of the many tripleo* packages. It's something RPMs can handle for you https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#_systemd

Reply:
No AFAIK they don't shipped as part of tripleo* pacakges, we run the services in containers and for managing those we create systemd units as part of roles itself https://github.com/openstack-archive/tripleo-ansible/blob/stable/wallaby/tripleo_ansible/roles/tripleo_container_manage/tasks/create.yml#L27